### PR TITLE
Update Travis CI for Coveralls

### PR DESCRIPTION
**Pull request by zfields**
### Problem

Travis CI is not integrated with Coveralls

### Solution

Update scripts, called by Travis CI, to generate Coveralls formatted coverage JSON, and submit the results to Coveralls.

### Steps to Test

Run Travis CI, and see if the coverage results appear in Coveralls

### References

[Coveralls](https://coveralls.io)
[Travis CI](https://travisci.org)

---

### Completeness

- [x] User is totes amazing for contributing!
- [x] Contributor has signed CLA ([Info here](https://github.com/spark/firmware/blob/develop/CONTRIBUTING.md))
- [x] Problem and Solution clearly stated
- [ ] Added to CHANGELOG.md after merging (add links to docs and issues)

### CODE REVIEW

File: ci/install_gcovr.sh [hunk 1]
```diff
@@ -1,14 +1,32 @@
...
+# Prerequisites:
+#   git
+#   python
+#   pip
 KERNEL_NAME=$(uname -s)
 
+# MacOS Support
 if [ "${KERNEL_NAME}" == "Darwin" ]; then
-  brew install gcovr || exit 1
+  brew install --force git python || exit 1
```
Comment (m-mcgowan) [inline ci/install_gcovr.sh]: why do we need --force? some comments would be useful

File: ci/install_gcovr.sh [hunk 2]
```diff
@@ -1,14 +1,32 @@
...
 else
   (apt-get -qq update &&
-  apt-get -qq install python-pip &&
-  pip install gcovr==4.1) || exit 2
+  apt-get -qq install git python-pip) || exit 1
+fi
+
+# Install custom `gcovr` to support Coveralls output
+pushd ~
+git clone https://github.com/zfields/gcovr.git -b coveralls
```
Comment (m-mcgowan) [inline ci/install_gcovr.sh]: I think it's normally best to keep device-os dependencies in a particle repo when we can. I realize your PR may be merged into the main gcovr repo, so we'll leave this as is for now. I've set myself a reminder to check back about this in 1 month.

File: ci/report/report.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
```
Comment (m-mcgowan) [inline ci/report/report.py]: Should this code have a particle copyright notice?

Comment (zfields) [reply ci/report/report.py]: It's not ours. I made improvements to make it work with Travis CI, but the original work is borrowed from the `cpp-coveralls` project.

I put it in it's own folder, along with it's LICENSE agreement, APL2.0, to highlight this fact.

Comment (m-mcgowan) [reply ci/report/report.py]: ok, thanks for clarifying!

**Pull request merged**